monkey/agent_plugins/credentials_collectors/chrome/src/linux_credentials_database_selector.py

```diff
@@ -1,11 +1,116 @@
-from typing import Collection
+import logging
+import os
+import pwd
+from pathlib import Path
+from typing import Collection, Dict, Sequence, Set
 
 from .utils import BrowserCredentialsDatabasePath
 
+logger = logging.getLogger(__name__)
+
+LinuxUsername = str
+UserDirectories = Dict[LinuxUsername, Path]
+
+DEFAULT_MASTER_KEY = "peanuts".encode()
+
+LOGIN_DATABASE_FILENAME = "Login Data"
+CHROMIUM_BASED_DB_PATHS = [
+    ("Google Chrome", ".config/google-chrome"),
+    ("Chromium", ".config/chromium"),
+    ("Brave", ".config/BraveSoftware/Brave-Browser"),
+    ("SlimJet", ".config/slimjet"),
+    ("Dissenter Browser", ".config/GabAI/Dissenter-Browser"),
+    ("Vivaldi", ".config/vivaldi"),
+    ("Microsoft Edge (Dev)", ".config/microsoft-edge-dev"),
+    ("Microsoft Edge (Beta)", ".config/microsoft-edge-beta"),
+    ("Microsoft Edge", ".config/microsoft-edge"),
+]
 
-class LinuxCredentialsDatabaseSelector:
-    def __init__(self):
-        pass
 
+class LinuxCredentialsDatabaseSelector:
     def __call__(self) -> Collection[BrowserCredentialsDatabasePath]:
-        return []
+        database_paths: Set[BrowserCredentialsDatabasePath] = set()
+        for profile_dir in self._get_browser_database_paths():
+            login_data_paths = self._get_login_data_paths(profile_dir)
+            database_paths.update(login_data_paths)
+
+        return database_paths
+
+    def _get_login_data_paths(
+        self, profile_dir_path: Path
+    ) -> Collection[BrowserCredentialsDatabasePath]:
+        login_data_paths: Set[BrowserCredentialsDatabasePath] = set()
+        try:
+            sub_profile_dirs = profile_dir_path.iterdir()
+            for subdir in sub_profile_dirs:
+                if subdir == (profile_dir_path / LOGIN_DATABASE_FILENAME):
+                    if self._file_is_accessible(subdir):
+                        login_data_paths.add(
+                            BrowserCredentialsDatabasePath(
+                                database_file_path=subdir, master_key=DEFAULT_MASTER_KEY
+                            )
+                        )
+
+                login_database_path = profile_dir_path / subdir / LOGIN_DATABASE_FILENAME
```

Comment:
Are all the files inside `profile_dir_path` only for the browser's user profiles? If no, this could add files to `login_data_paths` that aren't related to user profiles in any way.

Can't you just check the name of the directories and then add them to the set like we do for Windows? In Windows, we check for directories that start with "Profile".